Zend/zend_vm_def.h

```diff
@@ -451,7 +451,50 @@ ZEND_VM_COLD_CONSTCONST_HANDLER(16, ZEND_IS_IDENTICAL, CONST|TMP|VAR|CV, CONST|T
 	SAVE_OPLINE();
 	op1 = GET_OP1_ZVAL_PTR_DEREF(BP_VAR_R);
 	op2 = GET_OP2_ZVAL_PTR_DEREF(BP_VAR_R);
-	result = fast_is_identical_function(op1, op2);
+	if (Z_TYPE_P(op1) != Z_TYPE_P(op2)) {
+		/* They are not identical, return false. This has to check for __destruct errors (and undefined variable errors when IS_NULL is possible) */
+		FREE_OP1();
+		FREE_OP2();
+		ZEND_VM_SMART_BRANCH(0, 1);
+		return;
+	}
+	if (Z_TYPE_P(op1) <= IS_TRUE) {
+		/* They are identical, return true */
+		/* This has to check for undefined variable errors when IS_NULL is possible. */
+		FREE_OP1();
+		FREE_OP2();
+		ZEND_VM_SMART_BRANCH(1, 1);
```

Comment:
I guess this can probably be `((OP1_TYPE & IS_CV) || (OP2_TYPE & IS_CV))`, if only CVs can be undefined variables. http://nikic.github.io/2017/04/14/PHP-7-Virtual-machine.html#variable-types